Feature 날씨 앱 구현

woojump opened the pull request:
로딩 페이지 추가
- 기기의 위도와 경도 데이터를 얻습니다.
- openweathermap API를 통해 기기 위치의 날씨 및 대기 상태 데이터를 받아옵니다.

날씨 정보 페이지 추가
- 기기 위치를 바탕으로 현재 도시를 표시합니다.
- 현재 날짜와 시간을 표시합니다.
- 기기 위치의 기온과 날씨를 표시합니다.
- 기기 위치의 대기 상태(AQI, 미세먼지 농도, 초미세먼지 농도)를 표시합니다.

woojump committed:
test geolocator, http

woojump committed:
refactor

woojump committed:
add openweathermap api

woojump committed:
add weather_screen

woojump committed:
base structure of weather_screen

woojump committed:
build weather_screen part 1

woojump committed:
build weather_screen part 2

woojump committed:
build weather_screen part 3

woojump committed:
add loading animation

flutter-jihoon commented on lib/screens/loading.dart:
```diff
@@ -0,0 +1,73 @@
+}
+
+class _LoadingState extends State<Loading> {
+  late final double latitude;
+  late final double longitude;
+  late final dynamic weatherData;
+  late final dynamic airData;
```
이거 밑에서 쓸것도 아니면서 왜 클래스단에 정의되어 있나요?

flutter-jihoon commented on lib/data/my_location.dart:
```diff
@@ -0,0 +1,20 @@
+  Future<void> getMyCurrentLocation() async {
+    try {
+      // ignore: unused_local_variable
+      LocationPermission permission = await Geolocator.requestPermission();
```
여기 그냥 await한거 변수에 할당 안하면 린트 안걸리지 않나요?

flutter-jihoon commented on lib/data/network.dart:
```diff
@@ -0,0 +1,15 @@
+import 'dart:convert';
+
+class Network {
+  Future<dynamic> getJsonData(String url) async {
```
json이면 웬만하면 Map<String, dynamic>으로 명시해줍시다

flutter-jihoon commented on lib/models/model.dart:
```diff
@@ -0,0 +1,66 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+class Model {
```
클래스 이름이 너무 모호한 것 같습니다 무슨 모델인지 모르겠어요

flutter-jihoon commented on lib/screens/weather_screen.dart:
```diff
@@ -0,0 +1,272 @@
+    double doubleTemperature = weatherData['main']['temp'];
+    temperature = doubleTemperature.round();
+
+    Model model = Model();
```
이렇게 한 번만 객체 생성하는 클래스는 싱글톤 패턴을 활용하면 좋습니다

flutter-jihoon commented on lib/screens/weather_screen.dart:
```diff
@@ -0,0 +1,272 @@
+
+  @override
+  void initState() {
+    // TODO: implement initState
```
주석 불편합니다

flutter-jihoon commented on lib/screens/weather_screen.dart:
```diff
@@ -0,0 +1,272 @@
+        backgroundColor: Colors.transparent,
+        elevation: 0.0,
+        leading: IconButton(
+          onPressed: () {},
```
이거 뭐하는 버튼인가요

flutter-jihoon commented on lib/screens/weather_screen.dart:
```diff
@@ -0,0 +1,272 @@
+        ),
+        actions: [
+          IconButton(
+            onPressed: () {},
```
얘도

flutter-jihoon commented on lib/screens/weather_screen.dart:
```diff
@@ -0,0 +1,272 @@
+  var date = DateTime.now();
+  late int temperature;
+  late int weatherCondition;
+  late Widget weatherIcon;
+  late String description;
+  late int airCondition;
+  late Widget airIcon;
+  late String airState;
+  late double fineDust;
+  late double ultraFineDust;
+
+  @override
+  void initState() {
+    // TODO: implement initState
+    super.initState();
+    print(widget.parseAirData);
+    updateData(
+      weatherData: widget.parseWeatherData,
+      airData: widget.parseAirData,
+    );
+  }
+
+  void updateData({dynamic weatherData, dynamic airData}) {
+    cityName = weatherData['name'];
+    double doubleTemperature = weatherData['main']['temp'];
+    temperature = doubleTemperature.round();
+
+    Model model = Model();
+
+    weatherCondition = weatherData['weather'][0]['id'];
+    weatherIcon = model.getWeatherIcon(weatherCondition);
+    description = weatherData['weather'][0]['description'];
+
+    airCondition = airData['list'][0]['main']['aqi'];
+    airIcon = model.getAirIcon(airCondition);
+    airState = model.getAirState(airCondition);
+
+    fineDust = airData['list'][0]['components']['pm2_5'];
+    ultraFineDust = airData['list'][0]['components']['pm10'];
+  }
```
이거 지금은 문제없는데 맵에 키값 없으면 좀 위험하겠는데요

woojump committed:
Refactor

- 불필요한 주석을 제거하였습니다.
- 클래스 이름을 명확하도록 수정하였습니다.
- 불필요한 앱바를 제거하였습니다.
- 기타 코드를 개선했습니다.
